content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+---
+layout: post
+title: "Refactoring Conditional BuildSteps into Jenkins Pipeline"
+tags:
+- pipeline
+- freestyle
+- plugins
+- conditional-build-step
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:http://cloudbees.com[CloudBees].
+
+== Introduction
+
+With all the new developments in Jenkins Pipeline (and
+link:https://jenkins.io/blog/2017/01/12/declarative-pipeline-beta-2/[Declarative Pipeline on the horizon]),
+it's easy to forget what we did to create "pipelines" before
+*link:https://jenkins.io/doc/book/pipeline/[Pipeline]*.
+There are number of plugins, some that have been around since before Jenkins was
+Jenkins, that enable users to create "pipelines" in Jenkins.
+For example, basic job chaining worked well in many cases, and the
+link:https://plugins.jenkins.io/parameterized-trigger[Parameterized Trigger plugin]
+made chaining more flexible.
+However, creating chained jobs with conditional behavior was
+still one of the harder things to do in Jenkins.
+
+The
+link:https://plugins.jenkins.io/conditional-buildstep[Conditional BuildStep plugin]
+is a powerful tool that has allowed Jenkins users to write Jenkins jobs with
+complex conditional logic.  The plugin also includes a set of
+token macros that expose a wide range of job state information on which to base
```

Comment:
@daniel-beck 
I'm not sure what you mean by token macros are placeholders.
Conditional logic is composed of operations (called "conditions" in this case) and values.  The tokens provide the values.